### PR TITLE
feat: use ocis images from same directory level

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
FEAT: able to use images available at the same directory level as the markdown file in the ocis server.

Requirements:

>1. allow to embed images that are stored in the same folder or its subfolders by using the markdown image syntax

same folder - :heavy_check_mark:
subfolders - :x: (Follow up in next PR)

>2. not allowing to embed files from a folder of higher levels

:heavy_check_mark: 

>3. require the whole folder be publicly shared if the public viewer should see the images

:heavy_check_mark: 

>4. if only the markdown file is publicly share the viewer should not be able to access the images

:heavy_check_mark: 

### Test Cases:
- local images (image from current directory level)
- large local images
- multiple small images (if images are large then only one might be visible)
- uploaded image (upload into markdown)
- external image url
- audio/video (doesn't work - not supported by the md parser)

## Related Issue
- Part of #27

## Screenshots (if appropriate):
`Owner`:
Screenshot from 2024-04-15 12-11-10 | Screenshot from 2024-04-15 12-11-04
--|--

`Folder Link - Viewer`:
Screenshot from 2024-04-15 12-13-44 | Screenshot from 2024-04-15 12-13-52
--|--

`MD File link -  Editor`:
Screenshot from 2024-04-15 12-18-28 | Screenshot from 2024-04-15 12-12-26
--|--

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)
- [ ] Documentation only (no source changes)
- [ ] Maintenance (e.g. dependency updates or tooling)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation updated

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -59,9 +85,73 @@ onMounted(() => {
...
+    el.src = blobUrl
+    mediaUrls.value.push(blobUrl)
+  }
+}
+async function parseImageUrl(name: string) {
+  for (const file of unref(mediaFiles)) {
+    if (file.name === name) {
+      const url = await getUrlForResource(unref(currentFileContext).space, file)
+      // reload the active files
+      await loadFolderForFileContext(unref(currentFileContext))
```
Comment (saw-jan) [inline src/App.vue:145]: need caching. Will do a follow up PR

File: src/App.vue [hunk 2]
```diff
@@ -29,10 +35,27 @@ import 'reveal.js/dist/reveal.css'
...
-const dataSeparatorVertical = '\r?\n--\r?\n'
+import { getMediaMimeTypes } from './helpers/mediaMimeTypes'
+import { id as appId } from '../public/manifest.json'
+
 const themeStore = useThemeStore()
+const { loadFolderForFileContext, currentFileContext, activeFiles } = useAppDefaults({
+  applicationId: appId
+})
+const { getUrlForResource, revokeUrl } = useAppFileHandling({
+  clientService: useClientService
```
Comment (saw-jan) [inline src/App.vue:46]:
```suggestion
  clientService: useClientService()
```

File: src/App.vue [hunk 3]
```diff
@@ -29,10 +35,27 @@ import 'reveal.js/dist/reveal.css'
...
+import { id as appId } from '../public/manifest.json'
+
 const themeStore = useThemeStore()
+const { loadFolderForFileContext, currentFileContext, activeFiles } = useAppDefaults({
+  applicationId: appId
+})
+const { getUrlForResource, revokeUrl } = useAppFileHandling({
+  clientService: useClientService()
+})
+const appsStore = useAppsStore()
```
Comment (grgprarup) [inline src/App.vue:48]: image

Cannot load the presentation

Comment (saw-jan) [reply src/App.vue:48]: hmm, is `useAppsStore` only available on web-master?
Let me check

Comment (saw-jan) [reply src/App.vue:48]: Yeah, the code is not available in web-8.0. :cry:

Comment (saw-jan) [reply src/App.vue:48]: This PR will be merged to the main branch which will be v2.0.0.
We have release [v1.0.0](https://github.com/JankariTech/web-app-presentation-viewer/releases/tag/1.0.0)

**Comment by saw-jan** [comment]
@grgprarup @S-Panta I have updated the comments. Please re-check if it works

**Pull request merged**